Documentation/check-cmdref.sh

```diff
@@ -9,6 +9,7 @@ cmdref_dir="${script_dir}/cmdref"
 
 if ! git diff --quiet -- "${cmdref_dir}" ; then
     git --no-pager diff "${cmdref_dir}"
-    echo "HINT: to fix this, run 'make -C Documentation update-cmdref ; git commit Documentation/cmdref --message \"docs: Update cmdref\"'"
+    git checkout "${cmdref_dir}"
+    echo "HINT: to fix this, run 'make -C Documentation update-cmdref ; git commit Documentation/cmdref --signoff --message \"docs: Update cmdref\"'"
```

Comment:
I don't think this is fixing #10742:
```
$ git status
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   Documentation/cmdref/cilium-agent.md
	modified:   daemon/cmd/daemon_main.go

no changes added to commit (use "git add" and/or "git commit -a")
$
$ git diff
diff --git a/Documentation/cmdref/cilium-agent.md b/Documentation/cmdref/cilium-agent.md
index e1f915e6b..d644c0f41 100644
--- a/Documentation/cmdref/cilium-agent.md
+++ b/Documentation/cmdref/cilium-agent.md
@@ -66,7 +66,7 @@ cilium-agent [flags]
       --enable-k8s-event-handover                     Enable k8s event handover to kvstore for improved scalability
       --enable-l7-proxy                               Enable L7 proxy for L7 policy enforcement (default true)
       --enable-local-node-route                       Enable installation of the route which points the allocation prefix of the local node (default true)
-      --enable-node-port                              Enable NodePort type services by Cilium (beta)
+      --enable-node-port                              Enable NodePort type services by Cilium (beta) (default true)
       --enable-policy string                          Enable policy enforcement (default "default")
       --enable-remote-node-identity                   Enable use of remote node identity
       --enable-tracing                                Enable tracing while determining policy (debugging)
diff --git a/daemon/cmd/daemon_main.go b/daemon/cmd/daemon_main.go
index 855d53df4..c82e2e64c 100644
--- a/daemon/cmd/daemon_main.go
+++ b/daemon/cmd/daemon_main.go
@@ -497,7 +497,7 @@ func init() {
        flags.Bool(option.EnableHostPort, true, fmt.Sprintf("Enable k8s hostPort mapping feature (requires enabling %s)", option.EnableNodePort))
        option.BindEnv(option.EnableHostPort)
 
-       flags.Bool(option.EnableNodePort, false, "Enable NodePort type services by Cilium (beta)")
+       flags.Bool(option.EnableNodePort, true, "Enable NodePort type services by Cilium (beta)")
        option.BindEnv(option.EnableNodePort)
 
        flags.String(option.NodePortMode, option.NodePortModeHybrid, "BPF NodePort mode (\"snat\", \"dsr\", \"hybrid\")")
$
$ make
[...]
docker container run --rm --workdir /src/Documentation --volume /home/paul/cilium/Documentation/..:/src --user "1000:1000" cilium/docs-builder ./update-cmdref.sh
docker container run --rm --workdir /src/Documentation --volume /home/paul/cilium/Documentation/..:/src --user "1000:1000" cilium/docs-builder ./check-cmdref.sh
diff --git a/Documentation/cmdref/cilium-agent.md b/Documentation/cmdref/cilium-agent.md
index e1f915e6b..d644c0f41 100644
--- a/Documentation/cmdref/cilium-agent.md
+++ b/Documentation/cmdref/cilium-agent.md
@@ -66,7 +66,7 @@ cilium-agent [flags]
       --enable-k8s-event-handover                     Enable k8s event handover to kvstore for improved scalability
       --enable-l7-proxy                               Enable L7 proxy for L7 policy enforcement (default true)
       --enable-local-node-route                       Enable installation of the route which points the allocation prefix of the local node (default true)
-      --enable-node-port                              Enable NodePort type services by Cilium (beta)
+      --enable-node-port                              Enable NodePort type services by Cilium (beta) (default true)
       --enable-policy string                          Enable policy enforcement (default "default")
       --enable-remote-node-identity                   Enable use of remote node identity
       --enable-tracing                                Enable tracing while determining policy (debugging)
HINT: to fix this, run 'make -C Documentation update-cmdref ; git commit Documentation/cmdref --message "docs: Update cmdref"'
make[1]: *** [Makefile:27: check-cmdref] Error 1
make[1]: Leaving directory '/home/paul/cilium/Documentation'
make: *** [Makefile:473: postcheck] Error 2
```

Reply:
The output stays the same, but it doesn't leave modified `Documentation/cmdref` behind.

Reply:
Ah, I see. It used to be possible to build that way so maybe remove the `Fixes` tag from the commit message until #10742 is entirely addressed?

Reply:
I suspect that `make` always failed in this case, developers may have just worked around it by doing tricks like `make -C daemon` which skips the cmdref check target (@errordeveloper I think this is what you found as well when you looked?)

Reply:
I did double check before posting the review, but maybe I'm missing something.
```
$ git checkout 50aa04b
$ git log -n1
commit 50aa04bdb470f8ba9ba4ef574ef9c4b3d7fa949f (HEAD)
Author: André Martins <andre@cilium.io>
Date:   Wed Feb 26 09:35:09 2020 +0100

    golang: update to 1.14
    
    - Also, ran go fmt across some of the files as the go fmt in go 1.14
      changed
    
    Signed-off-by: André Martins <andre@cilium.io>
$
$ git status
HEAD detached at 50aa04bdb
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   Documentation/cmdref/cilium-agent.md
	modified:   daemon/daemon_main.go

no changes added to commit (use "git add" and/or "git commit -a")
$
$ git diff
diff --git a/Documentation/cmdref/cilium-agent.md b/Documentation/cmdref/cilium-agent.md
index 3987bad8b..68b4ea536 100644
--- a/Documentation/cmdref/cilium-agent.md
+++ b/Documentation/cmdref/cilium-agent.md
@@ -64,7 +64,7 @@ cilium-agent [flags]
       --enable-k8s-event-handover                             Enable k8s event handover to kvstore for improved scalability
       --enable-l7-proxy                                       Enable L7 proxy for L7 policy enforcement (default true)
       --enable-local-node-route                               Enable installation of the route which points the allocation prefix of the local node (default true)
-      --enable-node-port                                      Enable NodePort type services by Cilium (beta)
+      --enable-node-port                                      Enable NodePort type services by Cilium (beta) (default true)
       --enable-policy string                                  Enable policy enforcement (default "default")
       --enable-remote-node-identity                           Enable use of remote node identity
       --enable-tracing                                        Enable tracing while determining policy (debugging)
diff --git a/daemon/daemon_main.go b/daemon/daemon_main.go
index dbe30c754..715024d42 100644
--- a/daemon/daemon_main.go
+++ b/daemon/daemon_main.go
@@ -499,7 +499,7 @@ func init() {
                option.KubeProxyReplacementStrict, option.KubeProxyReplacementDisabled))
        option.BindEnv(option.KubeProxyReplacement)
 
-       flags.Bool(option.EnableNodePort, false, "Enable NodePort type services by Cilium (beta)")
+       flags.Bool(option.EnableNodePort, true, "Enable NodePort type services by Cilium (beta)")
        option.BindEnv(option.EnableNodePort)
 
        flags.String(option.NodePortMode, defaults.NodePortMode, "BPF NodePort mode (\"snat\", \"dsr\", \"hybrid\")")
$
$ make
[...]

build succeeded.

The dummy builder generates no files.
/home/paul/.local/lib/python3.7/site-packages/sphinx/application.py:402: RemovedInSphinx20Warning: app.info() is now deprecated. Use sphinx.util.logging instead.
  RemovedInSphinx20Warning)
/home/paul/.local/lib/python3.7/site-packages/sphinx/application.py:388: RemovedInSphinx20Warning: app.warning() is now deprecated. Use sphinx.util.logging instead.
  RemovedInSphinx20Warning)
make[2]: Leaving directory '/tmp/cilium/Documentation'
make[1]: Leaving directory '/tmp/cilium'
Build finished.
$
$ echo $?
0
```

Reply:
@pchaigno is it fixed for you?

Reply:
@aanm Nop. It works for you? :thinking:

Reply:
@pchaigno how come? your comment seem to give that impression, because of the result of `$?`

Reply:
Ah, I see. I wasn't clear. See the first two commands at the to: I'm at commit 50aa04bdb470f8ba9ba4ef574ef9c4b3d7fa949f, which predates this pull request and #10572. I just wanted to show that it used to work. It doesn't anymore and this pull request doesn't fix it.

Reply:
Ah, so you are starting with a modified tree, and in the past that hasn't failed the build, but now it does. Previously checks were done out of tree and now they are actually based on `git diff`. I'll look into it, personally I don't mind having to commit generated files, but does everyone else massively objects to that?

Reply:
I'm not a big fan of having to commit the changes, but I suspect I can just get away without it by using `make build` instead of `make` (which runs all the other checks that we want developers to run before they submit PRs).

Reply:
I've rebased and removed `Fixes: #10742` for now.